src/Apps/ArtAdvisor/07-Curated-Discovery/weaviate-db.ts

```diff
@@ -65,7 +73,46 @@ export class WeaviateDB {
       .withFields(
         "internalID slug title date rarity medium materials price dimensions imageUrl _additional { id distance }"
       )
-      .do()
+
+    if (priceMinUSD && priceMaxUSD) {
+      // range filter
```

Comment:
Range queries have a [different structure](https://weaviate.io/developers/weaviate/api/graphql/filters#multiple-operands) than simple ≤ or ≥ queries, thus this conditional logic.